File: python/program/mobil.py
# mobil.py
from kendaraan_bermotor import KendaraanBermotor
from mesin import Mesin
from transmisi import Transmisi
from tangki_bahan_bakar import TangkiBahanBakar
from ban import Ban

class Mobil(KendaraanBermotor):
    # constructor
    def __init__(self, merk="", model="", tipe_bodi="", jumlah_pintu=0,
                 daftar_ban=None, mesin=None, transmisi=None, tangki=None):
        super().__init__(merk, model, mesin, transmisi, tangki)
        self.tipe_bodi = tipe_bodi
        self.jumlah_pintu = jumlah_pintu
        self.daftar_ban = daftar_ban if daftar_ban is not None else []
        self.mesin = mesin if mesin else Mesin()
        self.transmisi = transmisi if transmisi else Transmisi()
        self.tangki_bahan_bakar = tangki if tangki else TangkiBahanBakar()

    # Setter
    def set_tipe_bodi(self, tipe_bodi):
        self.tipe_bodi = tipe_bodi

    def set_jumlah_pintu(self, jumlah_pintu):
        self.jumlah_pintu = jumlah_pintu

    def set_daftar_ban(self, daftar_ban):
        self.daftar_ban = daftar_ban

    def set_mesin(self, mesin):
        self.mesin = mesin

    def set_transmisi(self, transmisi):
        self.transmisi = transmisi

    def set_tangki_bahan_bakar(self, tangki):
        self.tangki_bahan_bakar = tangki

    # Getter
    def get_tipe_bodi(self):
        return self.tipe_bodi

    def get_jumlah_pintu(self):
        return self.jumlah_pintu

    def get_daftar_ban(self):
        return self.daftar_ban

    def get_mesin(self):
        return self.mesin

    def get_transmisi(self):
        return self.transmisi

    def get_tangki_bahan_bakar(self):
        return self.tangki_bahan_bakar

    # Method tambahan
    def tambah_ban(self, ban):
        self.daftar_ban.append(ban)

File: cpp/program/mobil.cpp
#pragma once
#include <iostream>
#include <vector>
#include "kendaraanBermotor.cpp"
#include "ban.cpp"

using namespace std;

class Mobil : public KendaraanBermotor {
private:
    // atribut
    string tipeBodi;
    int jumlahPintu;
    vector<Ban> daftarBan;
    Mesin mesin;
    Transmisi transmisi;
    TangkiBahanBakar tangkiBahanBakar;

public:
    // Constructor default
    Mobil() {}

    // Constructor parameter
    Mobil(string merk, string model, string tipeBodi, int jumlahPintu,
          vector<Ban> daftarBan, Mesin mesin,
          Transmisi transmisi, TangkiBahanBakar tangkiBahanBakar)
        : KendaraanBermotor(merk, model, mesin, transmisi, tangkiBahanBakar) 
    {
        this->tipeBodi = tipeBodi;
        this->jumlahPintu = jumlahPintu;
        this->daftarBan = daftarBan;
        this->mesin = mesin;
        this->transmisi = transmisi;
        this->tangkiBahanBakar = tangkiBahanBakar;
    }

    // Setter
    void setTipeBodi(string tipeBodi) { 
        this->tipeBodi = tipeBodi; 
    }

    void setJumlahPintu(int jumlahPintu) { 
        this->jumlahPintu = jumlahPintu; 
    }

    void setDaftarBan(vector<Ban> daftarBan) { 
        this->daftarBan = daftarBan; 
    }

    void setMesin(Mesin mesin) { 
        this->mesin = mesin; 
    }

    void setTransmisi(Transmisi transmisi) { 
        this->transmisi = transmisi; 
    }

    void setTangkiBahanBakar(TangkiBahanBakar tangkiBahanBakar) { 
        this->tangkiBahanBakar = tangkiBahanBakar; 
    }

    // Getter
    string getTipeBodi() { 
        return this->tipeBodi; 
    }
    
    int getJumlahPintu() { 
        return this->jumlahPintu; 
    }
    
    vector<Ban> getDaftarBan() { 
        return this->daftarBan; 
    }

    Mesin getMesin() { 
        return this->mesin; 
    }

    Transmisi getTransmisi() { 
        return this->transmisi; 
    }

    TangkiBahanBakar getTangkiBahanBakar() { 
        return this->tangkiBahanBakar; 
    }

    // Method tambahan
    void tambahBan(Ban b) { 
        daftarBan.push_back(b); 
    }

    // Destructor 
    ~Mobil() {}
};
